lib/project.js

```diff
@@ -25,8 +25,8 @@ Project.prototype = {
   version: Project$version,
   nameVer: Project$nameVer,
   persist: Project$persist,
-  optionalDep: Project$optionalDep,
-  script: Project$script,
+  optionalDep: getterSetter('optionalDependencies'),
+  script: getterSetter('scripts'),
```

Comment:
Stylistic preference: I am not a fan of this function name. Inside the func, you return `getSetPath` which at least gives a little more detail as to what you are getting and setting. This is a minor nitpick though.